machine-learning/02-openpbta_consensus_clustering.Rmd

```diff
@@ -0,0 +1,378 @@
+# exist yet
+results_dir <- "results"
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir, recursive = TRUE)
+}
+```
+
+#### Input files
+
+```{r input_files}
+histologies_file <- file.path(data_dir, "pbta-histologies-stranded-rnaseq.tsv")
+rnaseq_file <- file.path(data_dir, "pbta-vst-stranded.tsv")
+```
+
+#### Output files
+
+```{r output_files, live = TRUE}
+cc_results_file <- file.path(results_dir, "consensus_clustering_results.RDS")
+```
+
+## Read in and filter data
+
+### Sample metadata
+
+Let's read in the sample metadata and get the sample identifiers for medulloblastoma samples.
+
+```{r read_in_metadata, live = TRUE}
+# Read in metadata TSV
+histologies_df <- read_tsv(histologies_file)
+
+# Only pull out sample identifiers (KidsFirst biospecimen identifiers) that
+# correspond to medulloblastoma samples
+medulloblastoma_samples <- histologies_df %>%
+  filter(short_histology == "Medulloblastoma") %>%
+  pull(Kids_First_Biospecimen_ID)
+```
+
+### RNA-seq data
+
+```{r read_in_rnaseq}
+# Read in transformed RNA-seq data
+rnaseq_df <- read_tsv(rnaseq_file)
+
+# For our clustering validation analyses, we want a matrix of only 
+# medulloblastoma samples where the gene identifiers are rownames rather than
+# in the first column
+rnaseq_mat <- rnaseq_df %>%
+  # This makes sure we retain all of the columns with biospecimen IDs that 
+  # correspond to medulloblastoma samples
+  select(gene_id, all_of(medulloblastoma_samples)) %>%
+  tibble::column_to_rownames("gene_id") %>%
+  as.matrix()
+```
+
+## Cluster validation
+
+### Consensus clustering
+
+The first method we'll use is called consensus clustering.
+Consensus clustering to finds the "consensus" across multiple runs of the algorithm using a resampling procedure.
+
+We'll use the package [`ConsensusClusterPlus`](https://bioconductor.org/packages/release/bioc/html/ConsensusClusterPlus.html) that we loaded up top.
+
+The consensus clustering methodology was first introduced in [Monti et al. _Machine Learning_. 2003.](https://doi.org/10.1023/A:1023949509487)
+
+Consensus clustering is one way to help you determine the number of clusters in your data, but it is not the only methodology available. 
+Check out the [Data Novia course _Cluster Validation Essentials_ by Alboukadel Kassambara](https://www.datanovia.com/en/courses/cluster-validation-essentials/) for a deeper dive.
+
+```{r consensus_clustering}
+cc_results <- ConsensusClusterPlus(rnaseq_mat,
+                                   maxK = 15,
+                                   # Setting this seed is necessary for the 
+                                   # results to be reproducible
+                                   seed = 2020,
+                                   innerLinkage = "average",
+                                   finalLinkage = "average",
+                                   distance = "pearson")
+```
+
+It seems like there are 3 main stable clusters from the consensus clustering results.
```

Comment:
I feel like we are missing a bit of explanation of the expected output here. A bit more context about how you concluded 3 clusters might also be useful for people coming back to the notebook.